### ISSUE TITLE
connection retry behavior when dynamic authenticator is missing

### ISSUE THREAD

**Issue by sgoudelis** [crossbario/autobahn-js#443]
Hello,

I would like to clarify the behavior of AuthbahnJS when connection retries are enabled and when doing a Cryptosign authentication with a dynamic authenticator RPC registered by some backend component. What I am observing is this: if at any point while the AutobahnJS library is attempting to reconnect, the disconnection reason is 'wamp.error.no_such_procedure' (because the dynamic authenticator is not registered yet) there will never be another retry.

Is this the desired behavior ? Is there way to enable retries even in this case ?

**Comment by oberstet** [comment]
> if at any point while the AutobahnJS library is attempting to reconnect, the disconnection reason is 'wamp.error.no_such_procedure' (because the dynamic authenticator is not registered yet) there will never be another retry.

Yes, this is the case. The auto-reconnection logic only takes care to establish a WebSocket connection to the router. It is only then that the authentication opening handshake of WAMP is starting - and the logic for that considers `wamp.error.no_such_procedure` fatal

**Comment by jameshilliard** [comment]
> and the logic for that considers `wamp.error.no_such_procedure` fatal

Maybe there should be a flag/option that can be used to change this behavior.

**Comment by meejah** [comment]
On the Python side, we now have an `is_fatal` predicate that can be passed to the Component-style (re)connection object. So no error is considered "fatal" unless `is_fatal` returns `True` (and the default predicate is `return False`).

More context: when considering what is "a fatal error" it's really hard to come up with anything that's "definitely fatal" I think which is why I introduced the `is_fatal` callback/predicate in the first place. Even, e.g., "dns lookup failed" or "TLS cert is invalid" could at some point in the near future stop failing -- e.g. someone forgot to renew something.

**Comment by oberstet** [comment]
I think what @meejah came up for ABPy (https://github.com/crossbario/autobahn-python/blob/master/autobahn/wamp/component.py#L453) makes completely sense and IMO we should add a similar option (a user provided callback function) to ABJS.

**Comment by sgoudelis** [comment]
Are there any plans to support this in the near future ?
